source/analysis/REIXS/REIXSXESImageInterpolationAB.cpp

```diff
@@ -1210,3 +1180,18 @@ void REIXSXESImageInterpolationAB::setCorrelation2Smoothing(QPair<int,int> cSmoo
 	setModified(true);
 }
 
+void REIXSXESImageInterpolationAB::setBinningLevel(int binningLevel)
+{
+	binningLevel_ = binningLevel;
+	if(inputSource_){
+		axes_[0].size = inputSource_->size(0) / binningLevel_;
+		cachedValues_.resize(axes_.at(0).size);

```

Comment:
Unless you're wanting to preserve the values in these caches, I'd use `cachedValues_ = QVector<double>(axes_.at(0).size);` instead.